mmdet3d/datasets/once_dataset.py

```diff
@@ -0,0 +1,289 @@
+            - 'Camera': Box in camera coordinates.
+        filter_empty_gt (bool, optional): Whether to filter empty GT.
+            Defaults to True.
+        test_mode (bool, optional): Whether the dataset is in test mode.
+            Defaults to False.
+        pcd_limit_range (list, optional): The range of point cloud used to
+            filter invalid predicted boxes.
+            Default: [0, -40, -3, 70.4, 40, 0.0].
+    """
+    CLASSES = ('Car', 'Bus', 'Truck', 'Pedestrian', 'Cyclist')
+
+    def __init__(self,
+                 data_root,
+                 ann_file,
+                 split,
+                 pts_prefix='velodyne',
+                 pipeline=None,
+                 classes=None,
+                 modality=None,
+                 box_type_3d='LiDAR',
+                 filter_empty_gt=True,
+                 test_mode=False,
+                 pcd_limit_range=[0, -40, -3, 70.4, 40, 0.0],
+                 **kwargs):
+        super().__init__(
+            data_root=data_root,
+            ann_file=ann_file,
+            classes=classes,
+            pipeline=pipeline,
+            modality=modality,
+            box_type_3d=box_type_3d,
+            filter_empty_gt=filter_empty_gt,
+            test_mode=test_mode,
+            **kwargs)
+
+        self.split = split
+        self.root_split = os.path.join(self.data_root, split)
+        assert self.modality is not None
+        self.pcd_limit_range = pcd_limit_range
+        self.pts_prefix = pts_prefix
+
+        self.camera_list = ['cam01', 'cam03', 'cam05', 'cam06', 'cam07', 'cam08', 'cam09']
+        self.data_infos = list(filter(self._check_annos, self.data_infos))
+    
+    def __len__(self):
+        """Return the length of data infos.
+
+        Returns:
+            int: Length of filtered data infos.
+        """
+        return len(self.data_infos)
+    
+    def _check_annos(self, info):
+        return 'annos' in info
+
+    def get_data_info(self, index):
+        """Get data info according to the given index.
+
+        Args:
+            index (int): Index of the sample data to get.
+
+        Returns:
+            dict: Data information that will be passed to the data
+                preprocessing pipelines. It includes the following keys:
+
+                - sample_idx (str): Sample index.
+                - pts_filename (str): Filename of point clouds.
+                - img_prefix (str): Prefix of image files.
+                - img_filename (str, optional): image filename.
+                - lidar2img (list[np.ndarray], optional): Transformations
+                    from lidar to different cameras.
+                - ann_info (dict): Annotation info.
+        """
+        info = self.data_infos[index]
+        sample_idx = info['frame_id']
+        pts_filename = info['lidar_path']
+        
+        img_filenames = []
+        lidar2imgs = []
+        seq_id = info['sequence_id']
+        for camera in self.camera_list:
```

Comment:
Better to allow uses to only use 'lidar' data, like https://github.com/open-mmlab/mmdetection3d/blob/dev/mmdet3d/datasets/nuscenes_dataset.py#L222